packages/neo-one-client-core/src/__ledger_tests__/LedgerKeyStore.test.ts

```diff
@@ -0,0 +1,184 @@
+import { utils } from '@neo-one/utils';
+import BigNumber from 'bignumber.js';
+import _ from 'lodash';
+import { of } from 'rxjs';
+import { filter, take } from 'rxjs/operators';
+import { Ledger, LedgerKeyStore, LedgerProvider, Ledgers } from '../user/keystore/LedgerKeyStore';
+
+const testMessage =
+  '8000000185e7e907cc5c5683e7fc926ba4be613d1810aebe14686b3675ee27d2476e5201000002e72d286979ee6cb1b7e65dfddfb2e384100b8d148e7758de42e4168b71792c60a08601000000000013354f4f5d3f989a221c794271e0bb2471c2735ee72d286979ee6cb1b7e65dfddfb2e384100b8d148e7758de42e4168b71792c60e23f01000000000013354f4f5d3f989a221c794271e0bb2471c2735e';
+
+const flattenLedgers = (ledgers: Ledgers) =>
+  _.flatten(
+    Object.values(ledgers)
+      .filter(utils.notNull)
+      .map((networkLedgers) => Object.values(networkLedgers)),
+  ).filter(utils.notNull);
+
+describe('LedgerKeyStore', () => {
+  const emptyAccount = {
+    balances: {
+      ['dicarloin']: new BigNumber(0),
+    },
+  };
+
+  const activeAccount = {
+    balances: {
+      ['dicarloin']: new BigNumber(1),
+    },
+  };
+
+  const networks = ['test'];
+  const networks$ = of(networks);
+  const getAccount = jest.fn();
+
+  const ledgerProvider: LedgerProvider = {
+    networks$,
+    getAccount,
+    getNetworks: () => networks,
+  };
+
+  const indexAccount = {
+    type: 'ledger',
+    id: {
+      network: 'test',
+      address: 'ARL4VYQWxSwigpRsCsucLxovujKDmwYeTd',
+    },
+    name: '0',
+    publicKey: '03d81d462f80d55ff234b1b221e46d52e4da3f93c465edbbaeebd742bc1947fe6d',
+    configurableName: false,
+    deletable: false,
+  };
+
+  const secondaryAccount = {
+    type: 'ledger',
+    id: {
+      network: 'test',
+      address: 'AJGCVgoLp3bU57zd4k6mcpedWr45Em8dFD',
+    },
+    name: '1',
+    publicKey: '031546889cf12577237536380e86b2587d55743f9d4dac8879f332b79ce1a84cd6',
+    configurableName: false,
+    deletable: false,
+  };
+
+  const indexLedger: Ledger = {
+    accountKey: 0,
+    account: indexAccount,
+  };
+
+  const secondaryLedger: Ledger = {
+    accountKey: 1,
+    account: secondaryAccount,
+  };
+
+  let keystore: LedgerKeyStore;
+
+  beforeEach(() => {
+    getAccount.mockReset();
+    getAccount.mockImplementation(() => emptyAccount);
+    keystore = new LedgerKeyStore(ledgerProvider);
+  });
+
+  afterEach(() => {
```

Comment:
`afterEach(async () => {`

Reply:
https://jestjs.io/docs/en/api#aftereachfn-timeout

you don't need to async an `afterEach` if I'm understanding this right. Jest awaits any Promises in an afterEach on its own.